### ISSUE TITLE
Arrays in webhooks previous attributes parsed as Object[]

### ISSUE THREAD

**Issue by Flo354** [stripe/stripe-java#605]
Hi,

I am using stripe-java version 7.1.0 and I am currently experiencing something weird with Webhooks.
When getting `event.getData().getPreviousAttributes()`, I obtain a `Map<String, Object>` which is fine.

Within this map, if I am getting the field `verification`, is it also a `Map<String, Object>`.
But why an array, like `verification -> fields_needed` is an `Object[]` instead of a `List<Object>`.

Actually, I am asking this question to myself, because with my JSON library (Jackson JSON), When I am using this code: 

```java
String jsonStr = "some json";
Map<String, Object> result = mapper.readValue(jsonStr, Map.class);
``` 

When I am recovering  `verification -> fields_needed`, I am getting an `ArrayList<Object>`. And I find this more natural.

**Comment by ob-stripe** [comment]
Hi @Flo354, thanks for the report!

Stripe's Java library uses a custom deserializer for the `data` attribute of event objects. It is this custom deserializer that decodes JSON lists as Java arrays instead of `List` objects (cf. [here](https://github.com/stripe/stripe-java/blob/ebbc8bedb88e369d2103e67eb083e8bd42027152/src/main/java/com/stripe/model/EventDataDeserializer.java#L88-L97)).

I agree that `List` objets would feel more natural here and be more consistent with the rest of the API, so I'd be inclined to change the behavior here. Such a change would break backwards compatibility, but I think we were planning on releasing a new major version soon anyway, so we could bundle this change too.

@brandur-stripe @remi-stripe Any strong feels around this?

**Comment by remi-stripe** [comment]
I agree that List would make a lot more sense and I would tag as Future for when we do the next major revision.

**Comment by Flo354** [comment]
Thanks!

I'll wait for the next major release.

**Comment by ob-stripe** [comment]
Quick update: this didn't make it into the newly released major version 8.0.0, but it's still on our roadmap and will be fixed in a future major version.

**Comment by ob-stripe** [comment]
Fixed in stripe-java 9.0.0.
